pyproject.toml

```diff
@@ -88,7 +90,7 @@ ase = ["ase>=3.23.0"]
 # don't depend on tblite above 3.11 since unsupported https://github.com/tblite/tblite/issues/175
 tblite = ["tblite[ase]>=0.3.0; python_version<'3.12'"]
 vis = ["vtk>=6.0.0"]
-abinit = ["netcdf4>=1.6.5"]
+abinit = ["netcdf4>=1.7.1"]
```

Comment:
@janosh Janosh can I have a quick comment on this bump of `netcdf4` version here from commit 1dfc9e4f7bbddaf5d93fe9b09b425ca1be6a58f4?

It seems to cause CI failure (more details on https://github.com/materialsproject/pymatgen/pull/4128#discussion_r1810203743) in Ubuntu CI runner for `netcdf4>=1.7.1.post1` (1.7.1 is yanked), `1.6.5` seems to work though

I also tried to [install `delvewheel` but doesn't seem to help](https://github.com/materialsproject/pymatgen/pull/4128/commits/75e23b13b113f06e28bc6513a14485b0e0a0bb28)

There seem to be [a lot of discussion around this error](https://github.com/Unidata/netcdf4-python/issues?q=is%3Aissue%20state%3Aopen%20NetCDF%3A%20HDF%20error) but didn't see a solid solution yet. And the issue for me is that I cannot recreate this with my local Ubuntu machine (with exactly the same dependency versions).

Reply:
`netcdf` needed to be bumped in #3894 to fix the following CI errors:

```py
_____________ ERROR collecting tests/io/abinit/test_abiobjects.py _____________
tests\io\abinit\test_abiobjects.py:10: in <module>
    from pymatgen.io.abinit.abiobjects import (
src\pymatgen\io\abinit\__init__.py:5: in <module>
    from .netcdf import (
src\pymatgen\io\abinit\netcdf.py:25: in <module>
    import netCDF4
C:\Users\runneradmin\micromamba\envs\pmg\lib\site-packages\netCDF4\__init__.py:28: in <module>
    _delvewheel_patch_1_7_0()
C:\Users\runneradmin\micromamba\envs\pmg\lib\site-packages\netCDF4\__init__.py:25: in _delvewheel_patch_1_7_0
    raise OSError('Error loading {}; {}'.format(lib, ctypes.FormatError(ctypes.get_last_error())))
E   OSError: Error loading charset-[23](https://github.com/materialsproject/pymatgen/actions/runs/10265440264/job/28401647005#step:8:24)3f44715e5aacaf3a688c2faff5ddf7.dll; The operation completed successfully.
```

if it's causing problems again and this time downgrading helps, by all means let's do it :)

Reply:
Current `pymatgen` dependency declare has duplicate (would be fixed #4128), you must have been mislead too :)

The `netcdf4` bumped is not really the one installed in CI (the one in `optional` section is the one actually installed in CI, surely you know):
https://github.com/materialsproject/pymatgen/blob/3ee17e2c346dea6e9248eb04c989e58880914650/pyproject.toml#L95

https://github.com/materialsproject/pymatgen/blob/3ee17e2c346dea6e9248eb04c989e58880914650/pyproject.toml#L100-L113

So [it still installed](https://github.com/materialsproject/pymatgen/actions/runs/10307213885/job/28532023809):  `+ netcdf4==1.6.5`


I [looked into the error log again](https://github.com/materialsproject/pymatgen/actions/runs/10265440264/job/28401647005?pr=3894), and turns out it's `+ netcdf4==1.7.1.post1` causing the following error:
```
 ______________ ERROR collecting tests/io/abinit/test_pseudos.py _______________
tests\io\abinit\test_pseudos.py:11: in <module>
    from pymatgen.io.abinit.pseudos import Pseudo, PseudoTable
src\pymatgen\io\abinit\__init__.py:5: in <module>
    from .netcdf import (
src\pymatgen\io\abinit\netcdf.py:25: in <module>
    import netCDF4
C:\Users\runneradmin\micromamba\envs\pmg\lib\site-packages\netCDF4\__init__.py:28: in <module>
    _delvewheel_patch_1_7_0()
C:\Users\runneradmin\micromamba\envs\pmg\lib\site-packages\netCDF4\__init__.py:25: in _delvewheel_patch_1_7_0
    raise OSError('Error loading {}; {}'.format(lib, ctypes.FormatError(ctypes.get_last_error())))
E   OSError: Error loading charset-233f44715e5aacaf3a688c2faff5ddf7.dll; The operation completed successfully.
```

So I might revert that pin to `<1.7.1.post1` and ([1.7.1.post2 also triggers error](https://github.com/materialsproject/pymatgen/actions/runs/11471574769/job/31922707829?pr=4128)), [same for latest 1.7.2](https://github.com/materialsproject/pymatgen/actions/runs/11471711256/job/31923083563?pr=4128)